docs/protocol-drafts/BECKN-RFC-006-Signing-Beckn-APIs-In-HTTP-Draft-01.md

```diff
@@ -0,0 +1,409 @@
+
+`Signature keyId="example-bg.com|bg3456|ed25519",algorithm="ed25519",created="1641287885",expires="1641287885",headers="(created) (expires) digest",signature="hJ5sCmbe7s9Wateq6QAdBGloVSkLuLHWOXcRkzrMcVLthFldV4gnT9Vrnq9iDNPVSKuDqaercVjQwFlj0Ml+3Q=="`
+
+8. Finally the BG includes the X-Gateway-Authorization header in the request and calls the BPP search API
+
+**Note:**
+3. The difference between the `created` and the `expires` field should be equal to the TTL of the request.
+4. Also, the `expires` value should not be more than the expiration time of the key used for signing the request. If the expires value appears to be going beyond the lifespan of the signing key, generate a new key and update it on the registry via subscribe API OR use an existing registered key with an expiry time greater than the `expires` value of the signature
+
+
+
+### Step 4 : BPP verifies BG signature
+The BPP performs the following steps to authenticate the BAP and the BG and also ensure message integrity.
+
+1. Get keyId from the X-Gateway-Authorization header
+
+`example-bg.com|bg3456|ed25519`
+
+2. Split the keyID string using the delimiter `|`. As we know, the subscriber ID is the registered domain name of the subscriber and follows the format{subdomain}.{domain}.{extension}.
+3. The keyID uses the format {subscriber id}|{unique_public_key_id}|{signing algorithm} . If the signing algorithm extracted from the keyId does not match the value of the algorithm field in the X-Gateway-Authorization header, then the BPP should return an error.
+4. The keyID also contains a unique_public_key_id which is used when the BG has uploaded multiple public keys to a registry OR when the same domain is being used for implementing multiple subscribers.
+5. The BG will now look up the registry for the public key of the subscriber by sending the subscriber_id and the unique_key_id via the lookup API or by retrieving a cached copy of the subscriber's public key matching the subscriber_id and unique_key_id. It will get the BG’s public key :
+
+`The BG will now look up the registry for the public key of the subscriber by sending the subscriber_id and the unique_key_id via the lookup API or by retrieving a cached copy of the subscriber's public key matching the subscriber_id and unique_key_id. It will get the BG’s public key :`
+
+6. If no valid key is found, the BPP must return a NACK response with 401 Unauthorised response code.
+7. BPP will use the BG's public key to verify the signature. If the signature is verified, the BG is considered to be authenticated.
+8. If the signature is not verified, the BPP must return a NACK response with 401 Unauthorised response code with a Proxy-Authenticate header. For example, for an invalid BG signature, the BPP must return the following:
+
+**Headers:**
+
+HTTP/1.1 401 Unauthorized
+
+Proxy-Authenticate: Signature realm="example-bpp.com",headers="(created) (expires) digest"
+
+...
+
+**Request Body:**
+
```

Comment:
Can you please fix the code blocks using the "```" tags like this?

```
{
 "message": {
    "ack": {
        "status": "NACK"
    }
  }
}
```

Reply:
@BLR-0118